### PR TITLE
feat: add support for many new chains

### PR DESCRIPTION
**Pull request by daywiss**
Signed-off-by: David <david@umaproject.org>

**Motivation**
Add in support for many more chains

**Summary**
This adds support for OPTIMISM, METER, XDAI, SX, EVMOS,ARBITRUM, AVAX. Addresses are made explicit in this config to make it easy to double check for errors. All addresses were obtained from the [networks files](https://github.com/UMAprotocol/protocol/tree/master/packages/core/networks), except for goerli, which was obtained from a [yet to be merged pr](https://github.com/UMAprotocol/protocol/pull/4056). EDIT: john has provided the updated goerli addresses which no longer match this pr.

Pay special attention when reviewing to chain constants, as these had to be searched and investigated to find things like currency symbols, names, explorers.  Most currency logos are scaffolded with the eth logo, which should probably updated in a subsequent pr.  Also pay attention to oracle addresses to make sure those look correct. 

To enable or disable each chain, you must provide an rpc url for that chain in the env ( or remove to disable it).  I will add these envs to the production deployment (except for testnets).

### CODE REVIEW

File: src/helpers/oracleClient.ts [hunk 1]
```diff
@@ -42,6 +42,6 @@ forEach((client) => {
 
 if (process.env.REACT_APP_DEBUG) {
   events.on("change", (index, state) => {
-    console.log("event change", index, state);
+    console.log("change", index, state);
```
Comment (daywiss) [inline src/helpers/oracleClient.ts:45]: just reducing verbosity

File: src/components/index/createRequestsTableCells.tsx [hunk 2]
```diff
@@ -72,7 +72,11 @@ function createRequestsTableCells(
         ? req.proposedPrice.toString()
         : undefined;
 
-      if (proposedPrice && unanswerable.includes(proposedPrice)) {
+      // req.state 1 is fresh request with no proposed answer, in that case always set to "-"
+      // proposed price sometimes defaults to 0 even if noone has proposed an answer yet.
+      if (req.state === undefined || req.state <= 1) {
+        proposedPrice = "-";
```
Comment (daywiss) [inline src/components/index/createRequestsTableCells.tsx:78]: this was an unrelated bug, requests with no proposed answers would sometimes show that there was an answer of 0

File: src/components/index/createRequestsTableCells.tsx [hunk 3]
```diff
@@ -72,18 +73,19 @@ function createRequestsTableCells(
...
+      } else if (proposedPrice && unanswerable.includes(proposedPrice)) {
         proposedPrice = "Requested too early";
       } else if (
         proposedPrice &&
         proposedPrice !== "0" &&
         identifier !== "YES_OR_NO_QUERY"
```
Comment (daywiss) [inline src/components/index/createRequestsTableCells.tsx:85]: this was a bug, identifier was deceptively defined, it was actually the title of the request. This has been fixed above, there is now a title variable and the correct identifier value

File: src/components/request/useRequestTableData.tsx [hunk 4]
```diff
@@ -216,7 +216,7 @@ function useRequestTableData({
     });
 
     let umipLink =
-      "https://docs.umaproject.org/uma-tokenholders/approved-price-identifiers";
+      "https://docs.umaproject.org/resources/approved-price-identifiers";
```
Comment (daywiss) [inline src/components/request/useRequestTableData.tsx:219]: unrelated change, this url needed to be updated

File: src/constants/blockchain.ts [hunk 5]
```diff
@@ -72,7 +119,6 @@ export const CHAINS: Record<ChainId, ChainMetadata> = {
     name: "Polygon",
     chainId: ChainId.POLYGON,
     logoURI: polygonLogo,
-    rpcUrl: "https://polygon-rpc.com/",
```
Comment (pemulis) [inline src/constants/blockchain.ts:75]: Why drop this RPC URL?

File: src/constants/oracleConfig.ts [hunk 6]
```diff
@@ -20,81 +10,252 @@ const optimisticV2Chains: Record<
...
+    chainName: chain.name,
+  };
+  optimisticChains[chainId] = {
+    ...config,
     multicall2Address: "0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696",
     optimisticOracleAddress: "0xC43767F4592DF265B4a9F1a398B97fF24F38C6A6",
   };
+  skinnyChains[chainId] = {
+    ...config,
+    multicall2Address: "0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696",
```
Comment (pemulis) [inline src/constants/oracleConfig.ts:30]: Why is this address specified in debug mode but not elsewhere?

File: src/constants/oracleConfig.ts [hunk 7]
```diff
@@ -20,81 +10,252 @@ const optimisticV2Chains: Record<
...
+    blockExplorerUrls: [chain.explorerUrl],
+  };
+  // no skinny deployment on goerli
+  optimisticChains[chainId] = {
+    ...config,
+    optimisticOracleAddress: "0x6f26Bf09B1C792e3228e5467807a900A503c0281",
+  };
+  optimisticV2Chains[chainId] = {
+    ...config,
+    optimisticOracleAddress: "0x3C8a21099C202003Ec6f050Eb24F8f24a3828Ad3",
```
Comment (pemulis) [inline src/constants/oracleConfig.ts:79]: The newly deployed address is `0xA5B9d8a0B0Fa04Ba71BDD68069661ED5C0848884`.

My older PR was closed in favor of https://github.com/UMAprotocol/protocol/commit/f65eb4a256cc36c170af940cafa6ac7d6cae8fb1

Comment (pemulis) [inline src/constants/oracleConfig.ts:79]: New address is `0xA5B9d8a0B0Fa04Ba71BDD68069661ED5C0848884`

My old PR was closed in favor of https://github.com/UMAprotocol/protocol/commit/f65eb4a256cc36c170af940cafa6ac7d6cae8fb1

File: src/constants/oracleConfig.ts (hunk 6 above)
Comment (pemulis) [inline src/constants/oracleConfig.ts:30]: Why is this address only specified for debug mode?

File: src/constants/blockchain.ts (hunk 5 above)
Comment (pemulis) [inline src/constants/blockchain.ts:75]: This was removed since RPC URLs go in an env variable now?

**Review by pemulis — CHANGES_REQUESTED** [review]
Looks great! Commented with a correction on the Goerli OOv2 address, since my PR was superseded with another one. Otherwise just had some questions that are for my own information.

File: src/constants/oracleConfig.ts (hunk 6 above)
Comment (daywiss) [reply src/constants/oracleConfig.ts:30]: debug mode should be a fork of another chain, but internally we use chain id to identify known addresses, but we do not have it for the debug chain id (31337, or something likethat), since technically you could fork any chain, so you have to specify the addresses manually.

Comment (daywiss) [reply src/constants/oracleConfig.ts:30]: see previous answer

File: src/constants/blockchain.ts (hunk 5 above)
Comment (daywiss) [reply src/constants/blockchain.ts:75]: this was an unecessary value when copying the block from another code base.  the rpc url gets added through envs in the `oracleConfig` file

Comment (daywiss) [reply src/constants/blockchain.ts:75]: was an artifact of copy/paste, see other answer

File: src/constants/oracleConfig.ts (hunk 7 above)
Comment (daywiss) [reply src/constants/oracleConfig.ts:79]: good catch

File: src/constants/oracleConfig.ts (hunk 6 above)
Comment (daywiss) [reply src/constants/oracleConfig.ts:30]: i changed the environement variable to be separate from DEBUG ( which enables logs).

File: src/components/index/IndexComponent.tsx [hunk 8]
```diff
@@ -99,6 +103,11 @@ const Index = ({
       return result;
     }, initial);
     setFilteredRequests(nextFR);
+    if (process.env.REACT_APP_DEBUG && nextFR.old.length)
+      console.log(
+        `${nextFR.old.length} requests too old to show: `,
+        nextFR.old
+      );
```
Comment (daywiss) [inline src/components/index/IndexComponent.tsx:110]: added this because you may see requests on datastructure but not know why they are not showing up in UI. this will give you a hint at least if that happens

**Review by pemulis — APPROVED** [review]

**Comment by daywiss** [comment]
going to delay merging until the new sdk changes get published to bump the version here
